src/main/java/com/minecolonies/coremod/commands/citizencommands/ListCitizensCommand.java

```diff
@@ -59,37 +67,72 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
         return super.getCommandUsage(sender) + "<colonyId>";
     }
 
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final List<ActionArgument> actionArgumentList,
+            @NotNull final Map<String, Object> argumentValueByActionArgumentNameMap) throws CommandException
+    {
+        Integer page = null;
+        final Object pageObject = argumentValueByActionArgumentNameMap.get("page");
+        if (null != pageObject)
+        {
+            page = (Integer) pageObject;
+        }
+        Colony colony = null;
+        final Object colonyObject = argumentValueByActionArgumentNameMap.get("colony");
+        if (null != colonyObject)
+        {
+            colony = (Colony) colonyObject;
+        }
+        executeShared(server, sender, colony, page);
+    }
+
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
     {
         int colonyId = getIthArgument(args, 0, getColonyId(sender));
+        final Integer page = getIthArgument(args, 1, 1);
 
+        Colony colony = null;
         if (sender instanceof EntityPlayer)
         {
             if (colonyId == -1)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)